frontend_lib/src/component/EmojiReactions/EmojiReactionButtons.jsx

```diff
@@ -0,0 +1,252 @@
+        symbols: props.t('Symbols'),
+        flags: props.t('Flags'),
+        custom: props.t('Custom')
+      },
+      categorieslabel: props.t('Emoji categories'),
+      skintones: {
+        1: props.t('Default Skin Tone'),
+        2: props.t('Light Skin Tone'),
+        3: props.t('Medium-Light Skin Tone'),
+        4: props.t('Medium Skin Tone'),
+        5: props.t('Medium-Dark Skin Tone'),
+        6: props.t('Dark Skin Tone')
+      }
+    }
+
+    return (
+      <>
+        <button
+          title={props.t('React')}
+          ref={this.buttonRef}
+          className={classnames('ReactionButton__buttonpicker', { active: this.state.pickerDisplayed })}
+          onClick={() => this.setState(
+            prev => ({
+              pickerDisplayed: !prev.pickerDisplayed
+            })
+          )}
+        >
+          <i className='far fa-smile' />
+        </button>
+        {(this.state.pickerDisplayed
+          ? (
+            <UncontrolledPopover
+              className="EmojiPickerPopover"
+              toggle={props.onCancel} // eslint-disable-line react/jsx-handler-names
+              placement='left'
+              isOpen
+              target={() => this.buttonRef.current}
+            >
+              <PopoverBody>
+                <Picker
+                  native
+                  emojiTooltip={true}
+                  title=''
+                  i18n={emojiMartI18n}
+                  onSelect={this.handleSelect}
+                />
+              </PopoverBody>
+            </UncontrolledPopover>
+          )
+          : undefined
+        )}
+      </>
+    )
+  }
+}
+
+const TranslatedEmojiPickerButton = translate()(EmojiPickerButton)
+
+function getReactionButtonTitle (props) {
+  const reactionList = props.reactionList
+
+  const you = (
+    reactionList[0].reaction_id === props.userReactionId
+      ? props.t('You')
+      : props.t('you')
+  )
+
+  const value = props.reactionList[0].value
+  const emoji = getEmojiDataFromNative(value, 'apple', emojiData)
+
+  const users = andList(
+    props.reactionList.map(
+      ({ author, reaction_id: reactionId }) => (
+        reactionId === props.userReactionId
+          ? you
+          : (author.public_name || author.username)
+      )
+    )
+  )
+
+  const l10nOpts = {
+    reaction: emoji ? (emoji.id || emoji.name) : value,
+    users: users,
+    user: users
+  }
+
+  return (
+    props.userReactionId === -1
+      ? (
+        props.reactionList.length === 1
+          ? props.t('{{user}} reacted with {{reaction}}', l10nOpts)
+          : props.t('{{users}} reacted with {{reaction}}_them', l10nOpts)
+      )
+      : (
+        props.reactionList.length === 1
+          ? props.t('You reacted with {{reaction}}', l10nOpts)
+          : props.t('{{users}} reacted with {{reaction}}_you', l10nOpts)
+      )
+  )
+}
+
+const ReactionButton = translate()(function ReactionButton (props) {
+  const highlighted = props.userReactionId !== -1
```

Comment:
Wouldn't it be simpler if `highlighted` was directly given by the parent component?

Reply:
And `userReactionId` is not in `propTypes`.

Reply:
> Wouldn't it be simpler if highlighted was directly given by the parent component?

I don't think so: we need userReactionId anyway, and this would make `userReactionId` and `highlighted` a bit redundant. I'm also not sure this is the responsability of the parent component to control the look of the button and tell it to be hightlighted.

All that said, I'm not stuck on this and if you think it is simpler despite this, it most certainly is so we should go for it and I can pass an highlighted property.